### PR TITLE
Enhance left column in Vuln view

### PR DESCRIPTION
**Pull request by kgevorkyan**
Part of #2555

Capture

Upd:

Capture

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfoProps.kt [hunk 1]
```diff
@@ -140,6 +145,44 @@ val vulnerabilityGeneralInfoProps: FC<VulnerabilityGeneralInfoProps> = FC { prop
...
+                // ================= Related Vulnerabilities ===================
+
+                if (cosv.related?.isNotEmpty() == true) {
+                    hr { }
+                    h6 {
+                        className = ClassName("font-weight-bold text-primary-blue mb-4")
+                        +"Related Vulnerabilities".t()
+                    }
+                    p {
+                        +"${cosv.related}".drop(1).dropLast(1)
```
[the remark this answers is not in the document]
Comment (kgevorkyan) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfoProps.kt:182]: in aim to show array not `[1, 2, 3]` but `1, 2, 3`
more user friendly

[the remark this answers is not in the document]
Comment (nulls) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfoProps.kt:182]: `[1, 2, 3]` is array, and should be processed by JSON as a list. why it's a single string?

File: save-backend/src/test/resources/minio/cosv/5 [hunk 2]
```diff
@@ -21,6 +21,9 @@
     }
   ],
   "summary": "Incorrect Authorization",
+  "cwe_ids": ["1", "42", "308"],
```
Comment (nulls) [inline save-backend/src/test/resources/minio/cosv/5:24]: I guess `cwe_ids` is `CWE-<N>`, and `cwe_names` is short description.

like here https://cwe.mitre.org/top25/archive/2023/2023_stubborn_weaknesses.html

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfoProps.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (kgevorkyan) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfoProps.kt]: fixed

File: save-backend/src/test/resources/minio/cosv/5 (hunk 2 above)
Comment (kgevorkyan) [reply save-backend/src/test/resources/minio/cosv/5]: I though about it, but I was little bit confused by `CWE Name` and not `CWE Description`

changed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfoProps.kt [hunk 3]
```diff
@@ -112,7 +112,12 @@ val vulnerabilityGeneralInfoProps: FC<VulnerabilityGeneralInfoProps> = FC { prop
                     }
                     label {
                         className = ClassName("m-0")
-                        +(cosv.modified.toUnixCalendarFormat(TimeZone.currentSystemDefault()))
+                        buttonBuilder(
+                            cosv.modified.toUnixCalendarFormat(TimeZone.currentSystemDefault()),
```
[the remark this answers is not in the document]
Comment (kgevorkyan) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfoProps.kt]: changed to UTC

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfoProps.kt [hunk 4]
```diff
@@ -145,28 +145,31 @@ val vulnerabilityGeneralInfoProps: FC<VulnerabilityGeneralInfoProps> = FC { prop
...
+                if (cosv.cweIds?.isNotEmpty() == true) {
                     hr { }
                     h6 {
                         className = ClassName("font-weight-bold text-primary-blue mb-4")
-                        +"CWE Names".t()
+                        +"CWE IDs".t()
                     }
                     p {
-                        +"${cosv.cweNames}".drop(1).dropLast(1)
+                        +cosv.cweIds!!.joinToString(", ")
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfoProps.kt:157]:
```suggestion
                        +cosv.cweIds.orEmpty().joinToString(", ")
```

**Review by nulls — APPROVED** [review]

**Pull request merged**